core/solver/multigrid.cpp

```diff
@@ -212,17 +195,43 @@ struct MultigridState {
     /**
      * allocate_memory is a helper function to allocate the memory of one level
      *
-     * @tparam ValueType  the value type of memory
+     * @tparam VectorType  the vector type
      *
      * @param level  the current level index
      * @param cycle  the multigrid cycle
      * @param current_nrows  the number of rows of current fine matrix
      * @param next_nrows  the number of rows of next coarse matrix
      */
-    template <typename ValueType>
+    template <typename VectorType>
     void allocate_memory(int level, multigrid::cycle cycle,
                          size_type current_nrows, size_type next_nrows);
 
+#if GINKGO_BUILD_MPI
+    /**
+     * allocate_memory is a helper function to allocate the memory of one level
+     *
+     * @tparam VectorType  the vector type
+     *
+     * @param level  the current level index
+     * @param cycle  the multigrid cycle
+     * @param current_comm  the communicator of the current fine matrix
+     * @param next_comm  the communicator of the next coarse matrix
+     * @param current_nrows  the number of rows of the current fine matrix
+     * @param next_nrows  the number of rows of the next coarse matrix
+     * @param current_local_nrows  the number of rows of the local operator of
+     *                             the current fine matrix
+     * @param next_local_nrows  the number of rows of the local operator of the
+     *                          next coarse matrix
+     */
+    template <typename VectorType>
+    void allocate_memory(int level, multigrid::cycle cycle,
+                         experimental::mpi::communicator& current_comm,
+                         experimental::mpi::communicator& next_comm,
```

Comment:
Can't/Shouldn't they be const?
```suggestion
                         const experimental::mpi::communicator& current_comm,
                         const experimental::mpi::communicator& next_comm,
```

Reply:
good catch!